### ISSUE TITLE
"ENOENT: no such file or directory" after launching LLDB: Launch

### ISSUE THREAD

**Issue by John-Boccio** [jay-babu/mason-nvim-dap.nvim#113]
![image](https://github.com/jay-babu/mason-nvim-dap.nvim/assets/39712041/1345bdbc-3bb5-4a55-91b6-38288c05f46e)

Sorry if this is a dumb question. I am new to neovim and am starting from this kickstart.nvim repository here: https://github.com/John-Boccio/nvim

The changes I have made are in the main branch, all it consists of is the following:
```
│diff --git a/init.lua b/init.lua                                                                                                                                                                                                                                                  │
│index a7667dd..281434c 100644                                                                                                                                                                                                                                                     │
│--- a/init.lua                                                                                                                                                                                                                                                                    │
│+++ b/init.lua                                                                                                                                                                                                                                                                    │
│@@ -218,8 +218,8 @@ require('lazy').setup({                                                                                                                                                                                                                                       │
│   -- NOTE: Next Step on Your Neovim Journey: Add/Configure additional "plugins" for kickstart                                                                                                                                                                                    │
│   --       These are some example plugins that I've included in the kickstart repository.                                                                                                                                                                                        │
│   --       Uncomment any of the lines below to enable them.                                                                                                                                                                                                                      │
│-  -- require 'kickstart.plugins.autoformat',                                                                                                                                                                                                                                     │
│-  -- require 'kickstart.plugins.debug',                                                                                                                                                                                                                                          │
│+  require 'kickstart.plugins.autoformat',                                                                                                                                                                                                                                        │
│+  require 'kickstart.plugins.debug',                                                                                                                                                                                                                                             │
│                                                                                                                                                                                                                                                                                  │
│   -- NOTE: The import below can automatically add your own plugins, configuration, etc from `lua/custom/plugins/*.lua`                                                                                                                                                           │
│   --    You can use this folder to prevent any conflicts with this init.lua if you're interested in keeping                                                                                                                                                                      │
│diff --git a/lua/kickstart/plugins/debug.lua b/lua/kickstart/plugins/debug.lua                                                                                                                                                                                                    │
│index 7fc783f..1bab69b 100644                                                                                                                                                                                                                                                     │
│--- a/lua/kickstart/plugins/debug.lua                                                                                                                                                                                                                                             │
│+++ b/lua/kickstart/plugins/debug.lua                                                                                                                                                                                                                                             │
│@@ -39,6 +39,8 @@ return {                                                                                                                                                                                                                                                        │
│       ensure_installed = {                                                                                                                                                                                                                                                       │
│         -- Update this to ensure that you have the debuggers for the langs you want                                                                                                                                                                                              │
│         'delve',                                                                                                                                                                                                                                                                 │
│+        'codelldb',                                                                                                                                                                                                                                                              │
│+        'cppdbg',                                                                                                                                                                                                                                                                │
│       },                                                                                                                                                                                                                                                                         │
│     }    
```

After going into my basic C++ test project, I ran `:DapInstall codelldb`. Then I set a break point and launched "LLDB: Launch" and provided the path to my executable. The error is shown in the screen shot.

**Comment by nickmarks00** [comment]
I am having the same issue as of today, please re-open if possible

**Comment by jay-babu** [comment]
I can reopen but I don’t use codelldb, so I have no plans on investigating any time soon

**Comment by nickmarks00** [comment]
In case anyone comes across a similar issue, I resolved it by writing my own DAP configuration for `cpp`. It is as simple as following the steps described [here](https://github.com/mfussenegger/nvim-dap/wiki/C-C---Rust-(via--codelldb)). For me, doing this verbatim led to a working `codelldb` configuration.

**Comment by jay-babu** [comment]
@nickmarks00 can you create a PR with the diff?

**Comment by nickmarks00** [comment]
Hey, yes I can, I'll get onto that shortly ☺️

**Comment by nickmarks00** [comment]
So upon further investigation, there isn't actually any meaningful differences between the current configuration and the one I linked to. I am using lazy as my package manager, and so the actual issue was I had the plugin set to load on a command (`DapContinue`) rather than on an event like `VeryLazy`. This meant the order plugins were loaded once debugging started was probably awry, and hence why my problems were solved when I wrote my own adapter using the link supplied.
